oauth2/oauth2.go

```diff
@@ -146,6 +146,11 @@ func (c *Client) HttpClient() phttp.Client {
 	return c.hc
 }
 
+// set a specific auth method that can override the value set in NewClient
+func (c *Client) SetAuthMethod(authMethodValue string) {
```

Comment:
Per a chat with @russjones, let's make sure to explain why we're making the change here in a comment (and link to an issue if we have one). 

Context is that we'll eventually be upgrading to go-oidc v3 and we'll have to backport all of these fixes, so we'll need enough info to evaluate how & if this will need to be forward-ported when that happens.

Reply:
@timothyb89 please take a look at the updated comments

Reply:
and I added a GetAuthMethod